command/rootchain/server/server.go

```diff
@@ -133,7 +134,7 @@ func runRootchain(ctx context.Context, outputter command.OutputFormatter, closeC
 	}
 
 	// target directory for the chain
-	if err = os.MkdirAll(params.dataDir, 0700); err != nil {
+	if err = common.CreateDirSafe(params.dataDir, 0700); err != nil {
```

Comment:
Nitpick, maybe it is worth checking is it enough setting permissions to 0600 (I don't see that we need executable permission for this folder, since it just contains rootchain data).

```suggestion
	if err = common.CreateDirSafe(params.dataDir, 0600); err != nil {
```

Reply:
Executable permission on the folder does not mean you can execute executables within this folder. It allows entering into the directory (`cd`), and access inside of the folder. And to remove or add files to a directory both write and execute permissions are needed.

Reply:
My bad.